docker-compose.env.yml

```diff
@@ -0,0 +1,131 @@
+version: '3.4'
+
+# All environment variables referenced from
```

Comment:
I have 2 issues with this file:

1. It should not replicate _everything_ we have in the original `docker-compose.yml` file as maintaining parity between them would be a nightmare.
2. The list of environment variables getting out of sync with the original config file. Any idea how to simplify this?

Reply:
> 1. It should not replicate everything we have in the original docker-compose.yml file as maintaining parity between them would be a nightmare.

I think I'd mentioned in #439 only adding support for the connection information (host/port/db name) for the infra services... would that be acceptable? I think at the very least, avoiding configuration for anything that can be configured in-app is probably wise, yeah?

> 2. The list of environment variables getting out of sync with the original config file. Any idea how to simplify this?

I think in addressing point 1, that's probably mitigated since only core infrastructural config that's unlikely to change is exposed. What do you think?

Reply:
>I think I'd mentioned in #439 only adding support for the connection information (host/port/db name) for the infra services... would that be acceptable? I think at the very least, avoiding configuration for anything that can be configured in-app is probably wise, yeah?

Yup, that's easier. I'm wondering if we can somehow automate this. Worth thinking hard on it as that would make maintenance incredibly easier. Maybe establishing a pattern like `SENTRY_CONF_*` or something. Still though, I'm finding this harder to reason about the more implementation details I see:

 - If we need to list these in a file, then why not just list them in an `.env` file, with their values?
 - And then if we are listing them in a _file_, why can we not make that file the config file we already have?

I guess with tests + minimal connection options, this can be okay.